enterprise/cli/prebuilds_test.go

```diff
@@ -341,3 +354,144 @@ func TestPrebuildsSettingsAPI(t *testing.T) {
 		assert.False(t, settings.ReconciliationPaused)
 	})
 }
+
+// TestSchedulePrebuilds verifies the CLI schedule command when used with prebuilds.
+// Running the command on an unclaimed prebuild fails, but after the prebuild is
+// claimed (becoming a regular workspace) it succeeds as expected.
+func TestSchedulePrebuilds(t *testing.T) {
+	t.Parallel()
+
+	if !dbtestutil.WillUsePostgres() {
+		t.Skip("this test requires postgres")
+	}
+
+	// Setup
+	clock := quartz.NewMock(t)
+	clock.Set(dbtime.Now())
+	client, db, owner := coderdenttest.NewWithDatabase(t, &coderdenttest.Options{
+		Options: &coderdtest.Options{
+			DeploymentValues:         coderdtest.DeploymentValues(t),
+			IncludeProvisionerDaemon: true,
+			Clock:                    clock,
+		},
+		LicenseOptions: &coderdenttest.LicenseOptions{
+			Features: license.Features{
+				codersdk.FeatureWorkspacePrebuilds: 1,
+			},
+		},
+	})
+
+	cases := []struct {
+		name        string
+		cliErrorMsg string
+		cmdArgs     func(string) []string
+	}{
+		{
+			name:        "AutostartPrebuildError",
+			cliErrorMsg: "autostart configuration is not supported for prebuilt workspaces",
+			cmdArgs: func(workspaceName string) []string {
+				return []string{"schedule", "start", workspaceName, "7:30AM", "Mon-Fri", "Europe/Lisbon"}
+			},
+		},
+		{
+			name:        "AutostopPrebuildError",
+			cliErrorMsg: "autostop configuration is not supported for prebuilt workspaces",
+			cmdArgs: func(workspaceName string) []string {
+				return []string{"schedule", "stop", workspaceName, "8h30m"}
+			},
+		},
+		//{
+		//	name:        "ExtendPrebuildError",
```

Comment:
This test is failing because the workspace must have a deadline set in order to extend it. Currently, when a prebuild is claimed, its TTL is not set. This is addressed in PR: https://github.com/coder/coder/pull/19252/files#diff-4afb9f17accf40c5774d38a2fcec29fd71caeddca73dff6e37626623eebaaf50R10

Reply:
Unblocked after merge of https://github.com/coder/coder/pull/19252